Tools/TestWebKitAPI/Tests/WebKitCocoa/ScreenTime.mm

```diff
@@ -27,26 +27,32 @@
 
 #if ENABLE(SCREEN_TIME)
 
+#import "InstanceMethodSwizzler.h"
 #import "TestWKWebView.h"
+#import <ScreenTime/STWebHistory.h>
 #import <ScreenTime/STWebpageController.h>
 #import <WebKit/WKPreferencesPrivate.h>
+#import <WebKit/WKWebViewConfiguration.h>
 #import <WebKit/WKWebViewPrivate.h>
+#import <WebKit/WKWebsiteDataStore.h>
 #import <WebKit/_WKFeature.h>
+#import <pal/cocoa/ScreenTimeSoftLink.h>
 #import <wtf/RetainPtr.h>
 
 static void *blockedStateObserverChangeKVOContext = &blockedStateObserverChangeKVOContext;
 static bool stateDidChange = false;
 
-static RetainPtr<TestWKWebView> webViewForScreenTimeTests()
+static RetainPtr<TestWKWebView> webViewForScreenTimeTests(WKWebViewConfiguration *configuration = nil)
 {
-    RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    if (!configuration)
+        configuration = adoptNS([[WKWebViewConfiguration alloc] init]).autorelease();
+
     auto preferences = [configuration preferences];
     for (_WKFeature *feature in [WKPreferences _features]) {
         if ([feature.key isEqualToString:@"ScreenTimeEnabled"])
             [preferences _setEnabled:YES forFeature:feature];
     }
-    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 400, 300) configuration:configuration.get()]);
-    return webView;
+    return adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 400, 300) configuration:configuration]);
 }
 
 @interface STWebpageController ()
```

Comment:
I think you'll have to add `@property (nonatomic, copy) STWebHistoryProfileIdentifier profileIdentifier;` in here too, so this builds on older SDKs.